3.x: Adjust for zero token nodes

Bouncheck opened the pull request:
Adds ZeroTokenNodesIT that checks the behaviour of the driver when zero-token
nodes are involved.

Changes behavior of the driver in regards to zero-token nodes. Those nodes
will be ignored if encountered in `system.peers` table. If provided as a
contact point, the driver will attempt to connect to it, but will not populate
metadata with it. This results in zero-token nodes being not included in
query planning.

Note that the `isValidPeer` method had the check for nullity of `tokens` column
moved from extended peer check to the regular check.

dkropachev commented on driver-core/src/test/java/com/datastax/driver/core/ZeroTokenNodesIT.java:
```diff
@@ -0,0 +1,85 @@
+      minOSS = "6.2.0",
+      minEnterprise = "2024.2.2",
+      description = "Zero-token nodes introduced in scylladb/19684")
+  public void should_ignore_zero_token_peer() {
```
```suggestion
  public void should_ignore_single_zero_token_peer() {
```

dkropachev reviewed (changes requested):
@Bouncheck , thanks, looks good, could you please add another test for `Zero-token Datacenter` from https://github.com/scylladb/java-driver/issues/317

dkropachev commented:
@Bouncheck , please consider fixing code not to log following error:
```
WARN  [cluster1-worker-2] 2024-09-17 16:10:28,825 ReplicationStategy.java:204 - Error while computing token map for keyspace keyspace1 with datacenter eu-northscylla_node_north: could not achieve replication factor 3 (found 0 replicas only), check your keyspace replication settings.
```

Bouncheck commented:
Added some tests. The current version slightly differs from what's proposed in #317 , for example driver won't fail when executing queries after connecting to zero-token DC. It just won't direct queries to it.

Bouncheck commented:
> @Bouncheck , please consider fixing code not to log following error:
> 
> ```
> WARN  [cluster1-worker-2] 2024-09-17 16:10:28,825 ReplicationStategy.java:204 - Error while computing token map for keyspace keyspace1 with datacenter eu-northscylla_node_north: could not achieve replication factor 3 (found 0 replicas only), check your keyspace replication settings.
> ```

In one of the tests i create a table in a new keyspace with general replication_factor in multi-dc setting and I do not see this error now.

Bouncheck commented:
Some of the tests fail, because their hosts have `null` tokens but are not meant to be treated as zero-token nodes. This needs more or different adjustments.

Bouncheck committed:
Adjust for zero token nodes

Adds ZeroTokenNodesIT that checks the behaviour of the driver when zero-token
nodes are involved.

Changes behavior of the driver in regards to zero-token nodes. Those nodes
will be ignored if encountered in `system.peers` table. If provided as a
contact point, the driver will attempt to connect to it, but will not populate
metadata with it. This results in zero-token nodes being not included in
query planning.

Note that the `isValidPeer` method had the check for nullity of `tokens` column
moved from extended peer check to the regular check.
